### PR TITLE
Return self from DocSet operations.  No need to waste DocSets.

### PR DESCRIPTION
**Pull request by alexaryn**
This will allow code like this:
```
ds=context.read_binary(...)
ds.partition(...)
ds.merge(...)
ds.explode(...)
ds.embed(...)
ds.write.opensearch(...)
```

### CODE REVIEW

File: sycamore/docset.py
Comment (bohou-aryn) [inline sycamore/docset.py:1]: we do need to keep the docset plan tree read only for history, right?

Comment (alexaryn) [reply sycamore/docset.py:1]: The plan still is, right?  I'm open to discussion on how we want the code to work, but the existing behavior struck me as surprising and I couldn't see a technical reason in the current usage.  I wasn't so sure if there was a future vision I didn't get.

Comment (bohou-aryn) [reply sycamore/docset.py:1]: I think this immutability is pretty common in database execution layer. A lot of users just want do the following
ds1 = ds.partition()
ds2 = ds1.explode()
ds1.show()

we lose the track/ref to the plan in different stages.

**Comment by alexaryn** [comment]
It seems people like getting a new DocSet back from each method.  Closing.

**Pull request closed without merging**